algorithms/CPlusPlus/Trees/min_heap.cpp

```diff
@@ -0,0 +1,109 @@
+#include <iostream>
+#include <stdexcept>
+
+class Heap{
+	public:
+		void insert(const int& data);
+		void removemin();
+		void print();
+		Heap();
+
+	private:
+		int size_, capacity_;
+		int *items_;
+
+		int _parent(unsigned index) const;
+		int _child(unsigned index) const;
+
+		void _heapifyUP(unsigned index);
+		void _heapifyDOWN(unsigned index);
+		void _growArray();
```

Comment:
```suggestion
		int size, capacity;
		int *items;

		int parent(unsigned index) const;
		int child(unsigned index) const;

		void heapifyUP(unsigned index);
		void heapifyDOWN(unsigned index);
		void growArray();
```
using `_` as first char in variable name looks ugly

Reply:
isn't `_` the naming convention for private variable in c++?

Reply:
@Ashborn-SM looks like you are from C# background where people generally prefer using underscore as a prefix ( or suffix ) for private members of the the class to easily spot the scope of the variable when used and somewhat enhance the readability of the program.

But for C++, the case is different and can be quite disastrous especially when you use underscore as prefix while naming your identifier. It's because for C/C++, such names are reserved for language implementation to use and using it inside your code can cause conflicts with implementation variables leading to unwanted bugs which can be very hard to debug.

Reply:
thanks @Arsenic-ATG,
i will do the necessary changes